Students/Jacoby/Mechanum/subsystem/DriveTrain.java

```diff
@@ -0,0 +1,158 @@
+  private static CANSparkMax backRightSparkMax = new CANSparkMax(Constants.MOTOR_BACK_RIGHT, MotorType.kBrushless);
+
+
+  private static double[] motorCoefficients = {Constants.frontLeftMotorCoeff, Constants.frontRightMotorCoeff, Constants.backLeftMotorCoeff, Constants.backRightMotorCoeff};
+  
+  private static MecanumWrapperClass chassisDrive = new MecanumWrapperClass(frontLeftSparkMax, backLeftSparkMax, frontRightSparkMax, backRightSparkMax);
+
+  private static XboxController driverController = new XboxController(Constants.DRIVER_XBOX_CONTROLLER_PORT);
+
+  private static AHRS gyro = new AHRS();
+  private static double xInput;
+
+  private static double yInput;
+  private static double deltaZ;
+
+  private static double rInput;
+  
+  private static boolean brakeBooleanToggle = false;
+
+  private static int[] motorPolarity = {Constants.FRONT_LEFT_POLARITY, Constants.FRONT_RIGHT_POLARITY, Constants.BACK_LEFT_POLARITY, Constants.BACK_RIGHT_POLARITY};
+
+  public static void brakeModeToggle()
+
+  {
+    if(driverController.getBButtonReleased()){
+      brakeBooleanToggle = !brakeBooleanToggle;
+    }
+    if(brakeBooleanToggle){
+      frontLeftSparkMax.setIdleMode(CANSparkMax.IdleMode.kBrake);
+      backLeftSparkMax.setIdleMode(CANSparkMax.IdleMode.kBrake);
+      frontRightSparkMax.setIdleMode(CANSparkMax.IdleMode.kBrake);
+      backRightSparkMax.setIdleMode(CANSparkMax.IdleMode.kBrake);
+    }
+    else{
+      frontLeftSparkMax.setIdleMode(CANSparkMax.IdleMode.kCoast);
+      backLeftSparkMax.setIdleMode(CANSparkMax.IdleMode.kCoast);
+      frontRightSparkMax.setIdleMode(CANSparkMax.IdleMode.kCoast);
+      backRightSparkMax.setIdleMode(CANSparkMax.IdleMode.kCoast);
+    }
+ }
+ public static double getDeltaZ()
+ {
+   deltaZ = gyro.getDisplacementZ();
+
+   return deltaZ;
+ }
+
+ public static void stopMotors()
+ {
+   chassisDrive.driveCartesian
+   (driverController.getLeftX() * -1 * Constants.SPEED_MOD, 
+   driverController.getLeftY() * Constants.SPEED_MOD, 
+   driverController.getRightX() * -1 * Constants.SPEED_MOD,
+   new Rotation2d(), 
+   new double[] {0.0,0.0,0.0,0.0});
+ }
+
+ public static void driveForward(){
+    frontLeftSparkMax.set(-0.2);
+    frontRightSparkMax.set(0.2);
+    backLeftSparkMax.set(.2);
+    backRightSparkMax.set(-0.2);
+  }
+  public static void driveBackward(){
+    chassisDrive.driveCartesian(0, -.2, 0, null, motorCoefficients);
+  }
+  public static void driveLeft(){
+    chassisDrive.driveCartesian(.2, 0, 0, null, motorCoefficients);
+  }
+  public static void driveRight(){
+    chassisDrive.driveCartesian(-.2, 0, 0, null, motorCoefficients);
+  }
+  public static void driveAngleRight(){
+    chassisDrive.driveCartesian(0, 0, .2, null, motorCoefficients);
+  }
+  public static void driveAngeLeft(){
+    chassisDrive.driveCartesian(0, 0, -.2, null, motorCoefficients);
+  }
+
+
+
+
+  public static void driveMecanum()
```

Comment:
This function grabs controller input, but you did not use these values to drive the motors.